gateway/session_manager.go

```diff
@@ -230,7 +230,7 @@ func (sfr sessionFailReason) String() string {
 // sessionFailReason if session limits have been exceeded.
 // Key values to manage rate are Rate and Per, e.g. Rate of 10 messages
 // Per 10 seconds
-func (l *SessionLimiter) ForwardMessage(r *http.Request, currentSession *user.SessionState, rateLimitKey string, store storage.Handler, enableRL, enableQ bool, globalConf *config.Config, api *APISpec, dryRun bool) sessionFailReason {
+func (l *SessionLimiter) ForwardMessage(r *http.Request, currentSession *user.SessionState, rateLimitKey string, quotaKey string, store storage.Handler, enableRL, enableQ bool, globalConf *config.Config, api *APISpec, dryRun bool) sessionFailReason {
```

Comment:
do we need separate `quoteKey`?

Reply:
@jeffy-mathew yes. The problem is that in the case of a hashed key, quota and ratelimit compute the key differently. Ratelimit uses `currentSession.KeyHash()` always, and quota uses `storage.HashStr(currentSession.KeyID)`, but only in the case of a hashed key. Otherwise, it uses `currentSession.KeyID`. As a result, for ratelimit the key is computed "outside" of ForwardMessage function, while for quota it is computed "inside".

Since the ratelimit key is now passed from one place, this became a problem for the hashed keys. Hence, to fix itI had to add quotaKey to the ForwardMessage signature, so that the quota key is only applied when there is a custom ratelimit key specified in the session metadata.